src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,36 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        if (type == Bulldozer.class) {
+            return (List<T>) new BulldozerProducer().get();
+        } else if (type == Excavator.class) {
+            return (List<T>) new ExcavatorProducer().get();
+        } else if (type == Truck.class) {
+            return (List<T>) new TruckProducer().get();
+        } else {
+            return Collections.emptyList();
```

Comment:
When no producer has been found, return an empty list of the appropriate type instead of a raw `Collections.emptyList()`. This ensures that the method's return type is always consistent.